includes/payment-methods/class-wc-stripe-express-checkout-ajax-handler.php

```
@@ -91,8 +91,22 @@ public function ajax_add_to_cart() {
 			$product_id = isset( $_POST['product_id'] ) ? absint( $_POST['product_id'] ) : 0;
-			$qty        = ! isset( $_POST['qty'] ) ? 1 : absint( $_POST['qty'] );
-			$product    = wc_get_product( $product_id );
+			// Use wc_stock_amount() rather than absint() so stores that allow decimal
+			// quantities (e.g. selling fabric by the metre) keep fractional values like
+			// 0.25 instead of having them truncated to an integer. This mirrors how
+			// WooCommerce core parses the add-to-cart quantity.
+			// A malformed request can post qty as an array, and wc_stock_amount() (unlike
+			// the previous absint()) can return zero or a negative value. Reject both
+			// before touching the cart so we don't empty it and still report success.
+			if ( isset( $_POST['qty'] ) && is_array( $_POST['qty'] ) ) {
+				throw new Exception( __( 'Invalid product quantity.', 'woocommerce-gateway-stripe' ) );
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '102', column '15'). (undefined)

(MissingImport)